### ISSUE TITLE
MigrationInfo is not aware of the max column length of the flyway_schema_history table

### ISSUE THREAD

**Issue by olger** [nafg/slick-migration-api-flyway#26]
Hello,

If my analysis is correct, the description of the migration is compared during the migrate function.

I've ran in some issues because of an initial setup setup that combines a number of migrations into a version. It appears that the description is created based on migration.toString and that gives an issue in length (it goes over the 200 characters that appears to be the limit in the flyway table) and the comparison uses object references as the toString is not implemented to prevent this.

Example of the error:

```
org.flywaydb.core.api.FlywayException: Validate failed: 
Migration description mismatch for migration version 1
-> Applied to database : MigrationSeq(ArrayBuffer(slick.migration.api.TableMigration$Reversible@6415f61e, slick.migration.api.TableMigration$Reversible@334ebcaa, slick.migration.api.TableMigration$Reversible@5a0bef24, slick.m
-> Resolved locally    : MigrationSeq(ArrayBuffer(slick.migration.api.TableMigration$Reversible@61149fa5, slick.migration.api.TableMigration$Reversible@149c39b, slick.migration.api.TableMigration$Reversible@f2d890c, slick.mig
```

**Comment by nafg** [comment]
It doesn't seem like the issue is the length, but the different hash codes.

If you have a solution please open a PR

**Comment by olger** [comment]
Do different hash codes give the same exception message ?

I’ll try to provide a PR.

**Comment by olger** [comment]
I've added a PR, please don't merge it as it only a failing test to show the issue. 

As mentioned, I believe there is an issue with comparison of the description and the test at least show these symptoms. 

If this is really the issue, I'd prefer to have a descriptive piece of text inside description and like your thoughts how we can get there.

**Comment by olger** [comment]
I see some compilation errors in the current master due to 

`Error:(17, 12) class VersionedMigration needs to be abstract.
Missing implementations for 2 members. Stub implementations follow:
  def checksumMatches(x$1: Integer): Boolean = ???
  def checksumMatchesWithoutBeingIdentical(x$1: Integer): Boolean = ???
case class VersionedMigration[V](version: V, migration: Migration, info: MigrationInfo)`

I'll try to get these fixed but like to know what these methods are meant for. Especially the last one. Otherwise I may have wrong assumptions.

**Comment by olger** [comment]
BTW: If you are fine with the pull request and publish a new version, can you publish for 2.12 and 2.13 ? (many thanks for that :) )

**Comment by olger** [comment]
All in all its fixed in slick-migration-api version 0.8.0.
